src/material/datepicker/calendar-body.scss

```diff
@@ -31,7 +31,12 @@ $calendar-range-end-body-cell-size:
   padding-right: $calendar-body-label-side-padding;
 }
 
+.mat-calendar-body-cell-wrapper {
+  display: contents;
```

Comment:
Why is this necessary? Also it seems this isn't supported in Safari.

Reply:
Hmm, well, the [MDN Compatibility table for display](https://developer.mozilla.org/en-US/docs/Web/CSS/display#browser_compatibility) says there is some issues on Safari with unusual elements 🤔 . I tried this on Safari and it seems to render fine.

<img src="https://user-images.githubusercontent.com/7720245/146999470-fcfb1d49-72ca-4955-806d-7b9b5b553ee7.png" width=400/>
Safari Version 15.1 (17612.2.9.1.20)